### PR TITLE
[Documentation] pwa cloud install guide

### LINKED ISSUE #1226: [Documentation] PWA/Cloud Installation Guide Needed
Let's create a document on DevDocs that outlines how to install PWA with Magento Cloud. See the following as an initial reference: https://github.com/magento-research/magento2-upward-connector

### PR DESCRIPTION
**Pull request by jcalcaben**
## Description

Copy over the PWA install guide for Cloud into docs.

Also fixes broken links preventing doc build.

## Related Issue

Closes #1226 

## Verification Steps

1. In `pwa-devdocs` directory, run: `npm run lint src/tutorials/cloud-deploy/index.md`
2. Verify no linting errors.
3. Build HTML preview in `pwa-devdocs` directory (`npm run develop`)
4. Verify site builds successfully.
5. Navigate to `/tutorials/cloud-deploy`
6. Verify topic exists with content and is listed on the left nav

## How Have YOU Tested this?

See **Verification Steps**

## Screenshots / Screen Captures (if appropriate):

n/a

## Checklist:

- [x] I have updated the documentation accordingly, if necessary.
- [ ] I have added tests to cover my changes, if necessary.

### CODE REVIEW

File: pwa-devdocs/src/tutorials/cloud-deploy/index.md [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+"magento/sample-data-media-venia": "dev-master",
+```
+
+## Modify `.gitignore` file
+
+Add the following frontend dependency entries to the `.gitignore` file to track them in your repository.
+
+```text
+!package.json
+!yarn.lock
```
Comment (supernova-at) [inline pwa-devdocs/src/tutorials/cloud-deploy/index.md]: I've never seen the `!` in a `.gitignore`, what does it do?

File: pwa-devdocs/src/tutorials/cloud-deploy/index.md [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+"magento/module-configurable-sample-data-venia": "dev-master",
+"magento/module-customer-sample-data-venia": "dev-master",
+"magento/module-sales-sample-data-venia": "dev-master",
+"magento/module-tax-sample-data-venia": "dev-master",
+"magento/sample-data-media-venia": "dev-master",
+```
+
+## Modify `.gitignore` file
+
+Add the following frontend dependency entries to the `.gitignore` file to track them in your repository.
```
Comment (supernova-at) [inline pwa-devdocs/src/tutorials/cloud-deploy/index.md]: Adding entries to `.gitignore` makes it so that git _doesn't_ track them.

**Comment by supernova-at** [comment]
Failed verification on `npm run develop` step with the following error:

```
npm run develop

> pwa-devdocs@2.1.0-dev develop
> npm run build && node scripts/develop.js


> pwa-devdocs@2.1.0-dev build
> npm-run-all -s build-js build-css build-reference-doc-snippets build-compatibility-table build-html


> pwa-devdocs@2.1.0-dev build-js
> webpack --config webpack.config.js

Hash: 0de67e024d4d2d62203c
Version: webpack 4.29.0
Time: 86ms
Built at: 05/10/2019 5:01:33 PM
          Asset       Size  Chunks             Chunk Names
index.bundle.js  973 bytes       0  [emitted]  main
Entrypoint main = index.bundle.js
[0] multi ./src/_js/index.js 28 bytes {0} [built]
[1] ./src/_js/index.js 24 bytes {0} [built]

> pwa-devdocs@2.1.0-dev build-css
> node-sass --output-style compact --include-path src/_scss src/_scss/main.scss src/builds/css/main.css

Rendering Complete, saving .css file...
Wrote CSS to [redacted]

> pwa-devdocs@2.1.0-dev build-reference-doc-snippets
> node scripts/create-reference-docs

> Generating reference docs: src/_includes/auto-generated/peregrine/src/Price/Price.md
> Generating reference docs: src/_includes/auto-generated/peregrine/src/List/list.md

> pwa-devdocs@2.1.0-dev build-compatibility-table
> node scripts/create-compatibility-table.js


> pwa-devdocs@2.1.0-dev build-html
> bundle exec jekyll build --source src -I

sh: bundle: command not found
```

**Comment by jcalcaben** [comment]
@supernova-at You'll need to install Ruby and bundle dependencies 😞

File: pwa-devdocs/src/tutorials/cloud-deploy/index.md (hunk 1 above)
Comment (jcalcaben) [reply pwa-devdocs/src/tutorials/cloud-deploy/index.md]: It's a negation. So these specific files are guaranteed to be tracked.

File: pwa-devdocs/src/tutorials/cloud-deploy/index.md (hunk 2 above)
Comment (jcalcaben) [reply pwa-devdocs/src/tutorials/cloud-deploy/index.md]: The entries actually guarantees the file should be tracked.

File: pwa-devdocs/src/tutorials/cloud-deploy/index.md (hunk 1 above)
Comment (supernova-at) [reply pwa-devdocs/src/tutorials/cloud-deploy/index.md]: Isn't the same thing achieved if we just leave them out of `.gitignore`? Maybe we could drop this whole step?

**Comment by dpatil-magento** [comment]
@supernova-at I verified the steps and looks good.  
@tjwiebell Should this document be updated as per changes of #1213 for USE_FASTLY  0/1   Vs IMAGE_OPTIMIZING_ORIGIN  backend/onboard  ?

File: pwa-devdocs/src/tutorials/cloud-deploy/index.md (hunk 1 above)
Comment (jcalcaben) [reply pwa-devdocs/src/tutorials/cloud-deploy/index.md]: I think @bbatsche included this to make the point that these files are important and should be tracked.

Comment (bbatsche) [reply pwa-devdocs/src/tutorials/cloud-deploy/index.md]: If you take a look at the default `.gitignore` we use for cloud (https://github.com/magento/magento-cloud/blob/master/.gitignore) you'll see that it is kind of an inverse ignore file. By default, all files are excluded from git, and then specific files or directories are whitelisted back in

File: pwa-devdocs/src/tutorials/cloud-deploy/index.md [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+
+Add the repository information for the UPWARD PHP server and UPWARD Connector to the `repositories` section in your `composer.json`.
+
+```json
+"upward-connector": {
+    "type": "vcs",
+        "url": "https://github.com/magento-research/magento2-upward-connector"
+
+},
+  "upward-php": {
```
Comment (tjwiebell) [inline pwa-devdocs/src/tutorials/cloud-deploy/index.md]: Can remove `upward-php` as hard dependency now that it's published (this step was temporary).

File: pwa-devdocs/src/tutorials/cloud-deploy/index.md [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+          "url": "https://github.com/magento-research/upward-php"
+
+  }
+```
+
+To have composer install these packages, add them to the `require` section of the `composer.json` file.
+
+```json
+"magento/module-upward-connector": "^1.0.0",
+"magento/upward": "^1.0.0"
```
Comment (tjwiebell) [inline pwa-devdocs/src/tutorials/cloud-deploy/index.md]: Can remove `upward-php` as hard dependency now that it's published (this step was temporary).

**Pull request merged**